scipy/stats/_bws_test.py

```diff
@@ -0,0 +1,167 @@
+            By_num = (Hj - (m + n)/n * j) * abs(Hj - (m + n)/n * j)
+            Bx_den = i / (n + 1) * (1 - i / (n + 1)) * \
+                m * (m + n + 1) / (n + 2)
+            By_den = j / (m + 1) * (1 - j / (m + 1)) * \
+                n * (m + n + 1) / (m + 2)
+
+        Bx = 1 / n * np.sum(Bx_num / Bx_den)
+        By = 1 / m * np.sum(By_num / By_den)
+
+        B = (Bx + By) * 0.5
+    else:
+        if method == 'naive':
+            Bx_num = (Ri - (m + n + 1) / (n + 1) * i) ** 2
+            By_num = (Hj - (m + n) / m * j) ** 2
+        else:
+            Bx_num = (Ri - (m + n)/n * i) * abs(Ri - (m + n)/n * i)
+            By_num = (Hj - (m + n)/n * j) * abs(Hj - (m + n)/n * j)
+
+        Bx_den = i / (n + 1) * (1 - i / (n + 1)) * m * (m + n) / n
+        By_den = j / (m + 1) * (1 - j / (m + 1)) * n * (m + n) / m
+
+        Bx = 1 / n * np.sum(Bx_num / Bx_den)
+        By = 1 / m * np.sum(By_num / By_den)
+
+        if alternative == 'greater':
+            B = (Bx - By) * 0.5
+        else:
+            B = (By - Bx) * 0.5
+
+    return B
+
+
+BWSResult = namedtuple('BWSResult', ('statistic', 'pvalue'))
+
+
+@_axis_nan_policy_factory(BWSResult, n_samples=2)
+def bws_test(x, y, alternative="two-sided", method='naive'):
+    r'''Perform the Baumgartner-Weiss-Schindler test on two independent samples.
+
+    The Baumgartner-Weiss-Schindler (BWS) test is a nonparametric test of 
+    the null hypothesis that the distribution underlying sample `x` 
+    is the same as the distribution underlying sample `y`. Unlike 
+    the Kolmogorv-Smirnov, Wilcoxon, and Cramer-Von Mises tests, 
+    the BWS test weights the integral by the variance of the difference
+    in CDFs, emphasizing the tails of the distributions, which increases
+    the power of the test for a lot of applications.
+
+    Parameters
+    ----------
+    x, y : array-like
+        N-d arrays of samples. The arrays must be broadcastable except along
+        the dimension given by `axis`.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis. Default is 'two-sided'.
+        Let *F(u)* and *G(u)* be the cumulative distribution functions of the
+        distributions underlying `x` and `y`, respectively. Then the following
+        alternative hypotheses are available:
+
+        * 'two-sided': the distributions are not equal, i.e. *F(u) ≠ G(u)* for
+          at least one *u*.
+        * 'less': the distribution underlying `x` is stochastically less
+          than the distribution underlying `y`, i.e. *F(u) >= G(u)* for all *u*.
+        * 'greater': the distribution underlying `x` is stochastically greater
+          than the distribution underlying `y`, i.e. *F(u) <= G(u)* for all *u*.
+
+        Under a more restrictive set of assumptions, the alternative hypotheses
+        can be expressed in terms of the locations of the distributions;
+        see [3] section 5.1.
+    method : {'naive', 'variant'}, optional
```

Comment:
`method` usually (always) refers to the method used to calculate the p-value. This is more similar to the `variant` parameter of `kendalltau`. Can you use `variant` for the parameter and choose more descriptive strings for the possible inputs?

On the other hand, do we really need both? Are they both commonly used?

Reply:
The tricky thing is, different authors propose similar `variant` methods, as can be seen in p.5 [Murakami (2006)] and p.7 [Shan (2013)] at https://europepmc.org/backend/ptpmcrender.fcgi?accid=PMC4809145&blobtype=pdf
The way to calculate `B` statistics vary depending on which method you choose.

Reply:
He's suggesting that you name the argument `variant` instead of `method`. That there are multiple variants argues for his suggestion. `method='variant'` does not tell us which variant. `variant='shan'` does. But if you drop `'naive'` like most of us have been suggesting, you don't have to have this argument at all for this first version.